test/jdk/jdk/incubator/vector/templates/X-LoadStoreTest.java.template

```diff
@@ -268,7 +268,7 @@ public class $vectorteststype$ extends AbstractVectorLoadStoreTest {
 
     @DontInline
     static $abstractvectortype$ fromArray($type$[] a, int i) {
-        return $abstractvectortype$.fromArray(SPECIES, a, i);
+        return ($abstractvectortype$) SPECIES.fromArray(a, i);
```

Comment:
These new tests focus on the species method - but should we also test the statics factories in the record class, just in case a regression is introduced and the two implementation start diverging again?

Reply:
My expectation is the risk is small, but of course non-zero. These tests can be expensive to run so i was trying balance the risk without increasing test execution times.

I could strengthen the comment from:
```
        @ForceInline
        @Override final
        public ByteVector fromMemorySegment(MemorySegment ms, long offset, ByteOrder bo) {
            // User entry point:  Be careful with inputs.
            return ByteVector
                .fromMemorySegment(this, ms, offset, bo);
        }
```

to:

```
        @ForceInline
        @Override final
        public ByteVector fromMemorySegment(MemorySegment ms, long offset, ByteOrder bo) {
            // User entry point
            // Defer only to the equivalent method on the vector class, using the same inputs
            return ByteVector
                .fromMemorySegment(this, ms, offset, bo);
        }
```

Reply:
Sounds good

Reply:
Done